crates/sui-core/src/transaction_driver/mod.rs

```diff
@@ -88,41 +96,147 @@ where
+                if let Err(e) = result {
+                    tracing::info!("Timeout while driving ping transaction: {}", e);
+                }
+            }
+        }
+    }
+
+    /// Drives transaction to submission and effects certification. If ping is provided, then the requested will be treated as a ping transaction.
+    #[instrument(level = "error", skip_all, fields(tx_digest = ?request.transaction.as_ref().map(|t| t.digest()), ping = ?request.ping))]
     pub async fn drive_transaction(
         &self,
         request: SubmitTxRequest,
         options: SubmitTransactionOptions,
         timeout_duration: Option<Duration>,
     ) -> Result<QuorumTransactionResponse, TransactionDriverError> {
-        let tx_digest = request.transaction.digest();
-        let tx_type = if request.transaction.is_consensus_tx() {
-            TxType::SharedObject
+        // To determine the tx type, give priority to the provided ping type (if present).
+        // Also, for ping requests, the amplification factor is always 1.
+        let (tx_type, tx_digest, amplification_factor) = if let Some(ping) = request.ping {
+            let tx_type = if ping == PingType::FastPath {
+                TxType::SingleWriter
+            } else {
+                TxType::SharedObject
+            };
+            (tx_type, None, 1)
         } else {
-            TxType::SingleWriter
+            let transaction = request.transaction.as_ref().unwrap();
+
+            let tx_type = if transaction.is_consensus_tx() {
+                TxType::SharedObject
+            } else {
+                TxType::SingleWriter
+            };
+
+            let gas_price = transaction.transaction_data().gas_price();
+            let reference_gas_price = self.authority_aggregator.load().reference_gas_price;
+            let amplification_factor = gas_price / reference_gas_price.max(1);
+            if amplification_factor == 0 {
+                return Err(TransactionDriverError::ValidationFailed {
+                    error: UserInputError::GasPriceUnderRGP {
+                        gas_price,
+                        reference_gas_price,
+                    }
+                    .to_string(),
+                });
+            }
+            (tx_type, Some(*transaction.digest()), amplification_factor)
         };
 
-        let gas_price = request.transaction.transaction_data().gas_price();
-        let reference_gas_price = self.authority_aggregator.load().reference_gas_price;
-        let amplification_factor = gas_price / reference_gas_price.max(1);
-        if amplification_factor == 0 {
-            return Err(TransactionDriverError::ValidationFailed {
-                error: UserInputError::GasPriceUnderRGP {
-                    gas_price,
-                    reference_gas_price,
-                }
-                .to_string(),
-            });
-        }
-
+        let ping_label = request.ping.as_ref().map(|p| p.as_str()).unwrap_or("none");
```

Comment:
An alternative to consider: just set "false" vs "true" for the ping label, because the actual type is already in `tx_type`.